Makefile.base

```diff
@@ -106,7 +110,7 @@ else
   OBJDIR=$(OBJDIR_BASE)
 endif
 LOCAL_LDADD = -L$(TOPDIR)/src/lib/$(OBJDIR)
-COMP_FLAGS +=  -g -O2 -I/usr/include -I$(INCDIR) -I$(TOPDIR)/include -I. -Wall -fPIC 
+COMP_FLAGS +=  -g -O0 -I/usr/include -I$(INCDIR) -I$(TOPDIR)/include -I. -Wall -fPIC
```

Comment:
Hi Evan, the rest of the changes look good. We should definitely support no-optimization compiles, but for performance reasons, let's not change the default to -O0.
(When the OSX branch is in good shape, I want to merge it into master.)